client/src/components/aggregations/AggregationWidgetContainer.js

```diff
@@ -0,0 +1,215 @@
+  COUNT_PER_DATE: "countPerDate",
+  COUNT_PER_VALUE: "countPerValue",
+  NUMBERS_LIST: "numbersList",
+  VALUES_LIST: "valuesList",
+  OBJECTS_LIST: "objectsList",
+  LIKERT_SCALE_AND_PIE_AGG: "likertScaleAndPieAgg"
+}
+
+const DEFAULT_AGGREGATION_TYPE_PER_WIDGET_TYPE = {
+  [AGGERGATION_WIDGET_TYPE.LIKERT_SCALE]: AGGREGATION_TYPE.VALUES_LIST,
+  [AGGERGATION_WIDGET_TYPE.PIE]: AGGREGATION_TYPE.COUNT_PER_VALUE,
+  [AGGERGATION_WIDGET_TYPE.LIKERT_SCALE_AND_PIE]:
+    AGGREGATION_TYPE.LIKERT_SCALE_AND_PIE_AGG,
+  [AGGERGATION_WIDGET_TYPE.REPORTS_BY_TASK]: AGGREGATION_TYPE.REPORTS_BY_TASK,
+  [AGGERGATION_WIDGET_TYPE.REPORTS_MAP]: AGGREGATION_TYPE.OBJECTS_LIST,
+  [AGGERGATION_WIDGET_TYPE.IQR_BOX_PLOT]: AGGREGATION_TYPE.NUMBERS_LIST,
+  [AGGERGATION_WIDGET_TYPE.CALENDAR]: AGGREGATION_TYPE.COUNT_PER_DATE,
+  [AGGERGATION_WIDGET_TYPE.DEFAULT]: AGGREGATION_TYPE.VALUES_LIST
+}
+
+const AGGREGATION_TYPE_FUNCTION = {
+  [AGGREGATION_TYPE.REPORTS_BY_TASK]: reportsByTaskAggregation,
+  [AGGREGATION_TYPE.COUNT_PER_VALUE]: countPerValueAggregation,
+  [AGGREGATION_TYPE.COUNT_PER_DATE]: countPerDateAggregation,
+  [AGGREGATION_TYPE.NUMBERS_LIST]: numbersListAggregation,
+  [AGGREGATION_TYPE.VALUES_LIST]: valuesListAggregation,
+  [AGGREGATION_TYPE.OBJECTS_LIST]: objectsListAggregation,
+  [AGGREGATION_TYPE.LIKERT_SCALE_AND_PIE_AGG]: likertScaleAndPieAggregation
+}
+
+export const getAggregationWidget = (
+  fieldConfig,
+  defaultWidgetPerFieldType = DEFAULT_AGGREGATION_WIDGET_PER_FIELD_TYPE,
+  ignoreFieldConfigWidget = false
+) => {
+  const widget = !ignoreFieldConfigWidget && fieldConfig.aggregation?.widget
+  const defaultWidget = defaultWidgetPerFieldType[fieldConfig.type]
+  const defaultWidgetIsObject = typeof defaultWidget === "object"
+  return (
+    widget ||
+    (defaultWidget &&
+      defaultWidgetIsObject &&
+      defaultWidget[fieldConfig.widget]) ||
+    (!defaultWidgetIsObject && defaultWidget) ||
+    null
+  )
+}
+
+const getAggregationFunction = (fieldConfig, aggregationWidget) => {
+  const aggregationType =
+    fieldConfig.aggregation?.aggregationType ||
+    DEFAULT_AGGREGATION_TYPE_PER_WIDGET_TYPE[aggregationWidget]
+  return aggregationType
+    ? AGGREGATION_TYPE_FUNCTION[aggregationType] || null
+    : null
+}
+
+const AggregationWidgetContainer = ({
+  data,
+  fieldConfig,
+  fieldName,
+  period,
+  vertical,
+  widget,
+  widgetId,
+  ...otherWidgetProps
+}) => {
+  const WHEN_UNSPECIFIED = (
+    <div>
+      <em>Not specified</em>
+    </div>
+  )
+  const aggregationWidget = widget || getAggregationWidget(fieldConfig)
+  const aggregationFunction = getAggregationFunction(
+    fieldConfig,
+    aggregationWidget
+  )
+  if (!aggregationFunction) {
+    return null
+  }
+
+  const { values, ...otherAggregationDetails } = aggregationFunction(
+    fieldName,
+    fieldConfig,
+    data
+  )
+  const fieldProps = getFieldPropsFromFieldConfig(fieldConfig)
+  let label = fieldProps.label
+  if (label === undefined) {
```

Comment:
Perhaps add a comment explaining the difference between a `null` and an `undefined` label.